### ISSUE TITLE
Make Toc and MiniToc more accessible

### ISSUE THREAD

**Issue by 3y3** [diplodoc-platform/components#217]
MiniToc doesn't have role and states.
We need to add at minimum [nav](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/navigation_role) role and [current](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-current) state.

Also we need to think about MiniToc DOM position. Will be useful to add MiniToc before main content.

**Comment by martyanovandrey** [comment]
1) aria-current is processed by Scrollspy and sets the correct location
2) it is better not to add a title and navigation role, but to use the correct h1-h6 and nav tag for article and whole minitoc


![Image](https://github.com/diplodoc-platform/components/assets/37772440/0a65ae82-06f4-4bb8-a131-81892602ce09)

**Comment by martyanovandrey** [comment]
Same for Toc, need to use 
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/nav

And check how the behavior of the element will change after changing the div to nav.
https://stackoverflow.com/questions/14180785/html5-nav-element-vs-role-navigation

**Comment by martyanovandrey** [comment]
done

https://github.com/diplodoc-platform/components/commit/6f51f9c612af74dc6168e07d1f35d5ebb80dd905
https://github.com/diplodoc-platform/components/commit/3580bcbd3a92a917c152892a3e3575b9fbb69c4e
